gemma/gemini_client.cc

```diff
@@ -0,0 +1,382 @@
+    if (!api_key_.empty()) {
+      req.set_header("X-goog-api-key", api_key_);
+    }
+    req.body = request.dump();
+    
+    req.content_receiver = [&accumulated_response, this](const char* data, size_t data_length, uint64_t offset, uint64_t total_length) -> bool {
+        std::string chunk(data, data_length);
+        std::istringstream stream(chunk);
+        std::string line;
+        
+        while (std::getline(stream, line)) {
+          if (line.substr(0, 6) == "data: ") {
+            std::string event_data = line.substr(6);
+            
+            if (event_data == "[DONE]") {
+              if (!interactive_mode_) {
+                std::cout << "\n\n" << GREEN << "✅ Generation complete!" << RESET << std::endl;
+              }
+            } else {
+              try {
+                json event = json::parse(event_data);
+                if (event.contains("candidates") && !event["candidates"].empty()) {
+                  auto& candidate = event["candidates"][0];
+                  if (candidate.contains("content") && candidate["content"].contains("parts")) {
+                    for (const auto& part : candidate["content"]["parts"]) {
+                      if (part.contains("text")) {
+                        std::string text = part["text"].get<std::string>();
+                        std::cout << text << std::flush;
+                        accumulated_response += text;
+                      }
+                    }
+                  }
+                }
+              } catch (const json::exception& e) {
+                // Skip parse errors
+              }
+            }
+          }
+        }
+        return true;
+      };
+    
+    httplib::Response res;
+    httplib::Error error;
+    bool success = use_https_ ? ssl_client_->send(req, res, error) : client_->send(req, res, error);
+    
+    if (res.status == 200 && !accumulated_response.empty()) {
+      return json{
+        {"candidates", {{
+          {"content", {
+            {"parts", {{{"text", accumulated_response}}}}
+          }}
+        }}}
+      };
+    } else {
+      json error_response = {
+        {"error", {
+          {"message", "Streaming request failed"},
+          {"status", res.status}
+        }}
+      };
+      if (!res.body.empty()) {
+        error_response["error"]["details"] = res.body;
+      }
+      std::cerr << RED << "❌ Streaming request failed. Status: " << res.status << RESET << std::endl;
+      return error_response;
+    }
+  }
+
+private:
+  std::unique_ptr<httplib::Client> client_;
+  std::unique_ptr<httplib::SSLClient> ssl_client_;
+  std::string host_;
+  int port_;
+  std::string api_key_;
+  bool use_https_;
+  bool interactive_mode_;
+};
+
+void PrintUsage(const char* program) {
+  std::cout << "Usage: " << program << " [options]" << std::endl;
+  std::cout << "Options:" << std::endl;
+  std::cout << "  --host HOST      Server host (default: localhost)" << std::endl;
```

Comment:
Consider using our existing argument handling logic, perhaps a new struct? You can copy from LoaderArgs as a starting point.

Reply:
Added ClientArgs as the new struct.